File: registration-login-spring-boot-security-thymeleaf-rdbms/src/main/java/com/knf/dev/repository/UserRepository.java
package com.knf.dev.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.knf.dev.model.User;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
	User findByEmail(String email);
}

File: kotlin-spring-main/kotlin-spring-thymeleaf-login-signup/src/main/kotlin/com/knf/dev/demo/kotlinspringthymeleafloginsignup/repository/UserRepository.kt
package com.knf.dev.demo.kotlinspringthymeleafloginsignup.repository

import com.knf.dev.demo.kotlinspringthymeleafloginsignup.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User?, Long?> {
    fun findByEmail(email: String?): User?
}